### ISSUE TITLE
facter errors on unregistered system

### ISSUE THREAD

**Issue by vchepkov** [waveclaw/puppet-subscription_manager#94]
on servers not registered with a satellite fact puppet generates errors:

```
Error: Facter: error while resolving custom fact "rhsm_identity": Missing key or value to store
```

The problem is in code trying to cache a nil value:
https://github.com/waveclaw/puppet-subscription_manager/blob/master/lib/facter/rhsm_identity.rb#L66-L68

**Comment by vchepkov** [comment]
I have tried to submit a fix, but robocop is too picky :(

**Comment by waveclaw** [comment]
Should be resolved by work for #96.  Please retest with 5.4.1.

**Comment by vchepkov** [comment]
@waveclaw , I have tried version 5.4.1, works great, thank you!
